IFTPipeline.jl/src/preprocess.jl

```diff
@@ -296,5 +296,194 @@ function preprocess_single(; truecolor::T, falsecolor::T, landmask::T, landmask_
     @info "Writing segmented floes to $output"
     save_labeled_img(labeled_floes_cast, output)
 
+    return nothing
+end
+
+"""
+    preprocess_tiling_single(; truecolor::T, falsecolor::T, landmask::T, landmask_dilated::T, output::T) where {T<:AbstractString}
+
+Preprocess and segment floes in a single view. Save the segmented floes to `segmented` and the labeled floes to `labeled`.
+
+# Arguments
+- `truecolor`: path to truecolor image
+- `falsecolor`: path to falsecolor image
+- `landmask`: path to landmask image
+- `landmask_dilated`: path to dilated landmask image
+- `segmented`: path to segmented output file 
+- `labeled`: path to labeled output file 
+"""
+function preprocess_tiling_single(
+    ; 
+    truecolor::T, 
+    falsecolor::T, 
+    landmask_dilated::T, 
+    segmented::T,
+    labeled::T,
+
+    # Tiling parameters
+    tile_rblocks=8,
+    tile_cblocks=8,
+
+    # Ice labels thresholds
+    ice_labels_prelim_threshold=110.0,
+    ice_labels_band_7_threshold=200.0,
+    ice_labels_band_2_threshold=190.0,
+    ice_labels_ratio_lower=0.0,
+    ice_labels_ratio_upper=0.75,
+
+    # Adaptive histogram equalization parameters
+    adapthisteq_white_threshold=25.5, 
+    adapthisteq_entropy_threshold=4, 
+    adapthisteq_white_fraction_threshold=0.4,
+
+    # Gamma parameters
+    gamma=1,
+    gamma_factor=1,
+    gamma_threshold=220,
+
+    # Unsharp mask parameters
+    unsharp_mask_radius=10, 
+    unsharp_mask_amount=2.0, 
+    unsharp_mask_factor=255.0,
+
+    # Brighten parameters
+    brighten_factor = 0.1,
+
+    # Preliminary ice mask parameters
+    prelim_icemask_radius=10, 
+    prelim_icemask_amount=2, 
+    prelim_icemask_factor=0.5,
+
+    # Main ice mask parameters
+    icemask_band_7_threshold=5,
+    icemask_band_2_threshold=230,
+    icemask_band_1_threshold=240,
+    icemask_band_7_threshold_relaxed=10,
+    icemask_band_1_threshold_relaxed=190,
+    icemask_possible_ice_threshold=75,
+    icemask_n_clusters=3,
+
+    ) where {T<:AbstractString}
+
+    @info "Processing images: $truecolor, $falsecolor, $landmask_dilated"
+    truecolor_img = loadimg(; dir=dirname(truecolor), fname=basename(truecolor))
+    falsecolor_img = loadimg(; dir=dirname(falsecolor), fname=basename(falsecolor))
+
+    # TODO: make symmetric landmask saving/loading functions
+    landmask_dilated=BitMatrix(FileIO.load(landmask_dilated))
+    
+    # Invert the landmasks – in the tiling version of the code, 
+    # the landmask is expected to be the other polarity compared with
+    # the non-tiling version.
+    landmask = (
+        dilated=map(!, landmask_dilated),
+    )
+
+    @info "Remove alpha channel if it exists"
+    rgb_truecolor_img = RGB.(truecolor_img)
+    rgb_falsecolor_img = RGB.(falsecolor_img)
+
+    @info "Get tile coordinates"
+    tiles = IceFloeTracker.get_tiles(
+        rgb_truecolor_img; 
+        rblocks=tile_rblocks, 
+        cblocks=tile_cblocks
+    )
```

Comment:
This is the part that would need to be changed to enable support for tile size. The rest is the same.